File: 187_repeated_DNA_sequences/findRepeatedDnaSequence.py
class Solution(object):
    def findRepeatedDnaSequences(self, s):
        """
        : type s: str
        : rtype List[str]
        """
        Result = []
        if len(s) < 10:
            return Result

        Str_dic = {}
        idx = 0
        while (idx + 10) <= len(s):
            if Str_dic.has_key(s[idx:idx+10]):
                Str_dic[s[idx:idx+10]] += 1
            else:
                Str_dic[s[idx:idx+10]] = 1            
            idx += 1

        Result = [item[0] for item in Str_dic.items() if item[1] > 1]

        return Result



if __name__ == '__main__':
    sol = Solution()
    s = "AAAAACCCCCAAAAACCCCCCAAAAAGGGTTT"
    s1 = ''
    print sol.findRepeatedDnaSequences(s)

File: 187_repeated_DNA_sequences/findRepeatedDnaSequence.cpp
#include <map>
#include <vector>
#include <string>
#include <iostream>

using namespace std;

class Solution {
	public:
    	vector<string> findRepeatedDnaSequences(string s) {
            map<string, int> dict;
            vector<string> result;
            int i = 0;
            while(i + 10 <= s.size())
            {
                string sub = s.substr(i, 10);
                if(++dict[sub] > 1)
                {
                    result.push_back(sub);
                    dict[sub] = INT_MIN;
                }
                i++;
            }
            return result;
    	}
};
int main()
{
    string s = "AAAAACCCCCAAAAACCCCCCAAAAAGGGTTT";
    Solution sol;
    vector<string> result;
    result = sol.findRepeatedDnaSequences(s);
    for(int i = 0; i < result.size(); i++)
    {
        cout << result[i] << endl;
    }
}
